src/sentry/grouping/ingest.py

```diff
@@ -581,3 +581,17 @@ def check_for_category_mismatch(group: Group) -> bool:
 
 def extract_hashes(calculated_hashes: CalculatedHashes | None) -> list[str]:
     return [] if not calculated_hashes else list(calculated_hashes.hashes)
+
+
+def project_uses_optimized_grouping(project: Project) -> bool:
+    primary_grouping_config = project.get_option("sentry:grouping_config")
+    secondary_grouping_config = project.get_option("sentry:secondary_grouping_config")
+    has_mobile_config = "mobile:2021-02-12" in [primary_grouping_config, secondary_grouping_config]
+
+    return (
+        features.has(
+            "organizations:grouping-suppress-unnecessary-secondary-hash",
+            project.organization,
+        )
+        and not has_mobile_config
+    )
```

Comment:
Short circuit a feature lookup if it's not necessary:

```suggestion
    if has_mobile_config:
        return False
    return features.has(
            "organizations:grouping-suppress-unnecessary-secondary-hash",
            project.organization,
        )
```

Reply:
Ah, good idea.